widgets/numeric-refinement-list/numeric-refinement-list.js

```diff
@@ -0,0 +1,184 @@
+ * @param  {string|Function} [options.templates.footer] Footer template
+ * @param  {Function} [options.transformData] Function to change the object passed to the item template
+ * @param  {boolean} [hideContainerWhenNoResults=true] Hide the container when there's no results
+ * @return {Object}
+ */
+function numericRefinementList({
+  container,
+  attributeName,
+  options,
+  cssClasses: userCssClasses = {},
+  templates = defaultTemplates,
+  transformData,
+  hideContainerWhenNoResults = true
+  }) {
+  let containerNode = utils.getContainerNode(container);
+  let usage = 'Usage: numericRefinementList({container, attributeName, options, [sortBy, limit, cssClasses.{root,header,body,footer,list,item,active,label,checkbox,count}, templates.{header,item,footer}, transformData, hideContainerWhenNoResults]})';
+
+  let RefinementList = headerFooter(require('../../components/RefinementList/RefinementList.js'));
+  if (hideContainerWhenNoResults === true) {
+    RefinementList = autoHideContainer(RefinementList);
+  }
+
+  if (!container || !attributeName) {
+    throw new Error(usage);
+  }
+
+  return {
+    getConfiguration: () => {
+      return {};
+    },
+
+    render: function({results, helper, templatesConfig, state, createURL}) {
+      let templateProps = utils.prepareTemplateProps({
+        transformData,
+        defaultTemplates,
+        templatesConfig,
+        templates
+      });
+
+      let facetValues = options.map(function (option) {
+        option.isRefined = isRefined(helper.state, attributeName, option);
+        return option;
+      });
+
+      let hasNoResults = facetValues.length === 0;
+
+      let cssClasses = {
+        root: cx(bem(null), userCssClasses.root),
+        header: cx(bem('header'), userCssClasses.header),
+        body: cx(bem('body'), userCssClasses.body),
+        footer: cx(bem('footer'), userCssClasses.footer),
+        list: cx(bem('list'), userCssClasses.list),
+        link: cx(bem('link'), userCssClasses.link),
+        active: cx(bem('item', 'active'), userCssClasses.active)
+      };
+
+      ReactDOM.render(
+        <RefinementList
+          createURL={(facetValue) => createURL(refine(state, attributeName, options, facetValue))}
+          cssClasses={cssClasses}
+          facetValues={facetValues}
+          shouldAutoHideContainer={hasNoResults}
+          templateProps={templateProps}
+          toggleRefinement={toggleRefinement.bind(null, helper, attributeName, options)}
+          />,
+        containerNode
+      );
+    }
+  };
+}
+
+function isRefined(state, attributeName, option) {
+  var currentRefinements = state.getNumericRefinements(attributeName);
+
+  if (option.start !== undefined && option.end !== undefined) {
+    if (option.start === option.end) {
+      return currentRefinements['='] != undefined &&
+        currentRefinements['='].find(function (refinement) { return refinement == option.start }) !== undefined;
```

Comment:
Yes and you can also use the ES6 fat arrow syntax `refinement => return retinfment === option.start`.